src/commands/test/lib/wizard/uitest.ts

```diff
@@ -0,0 +1,152 @@
+import * as fs from "fs";
+import * as path from "path";
+import { CommandArgs, CommandResult, AppCommand } from "../../../../util/commandline";
+import { AppCenterClient } from "../../../../util/apis";
+import { prompt, out } from "../../../../util/interaction";
+import RunUITestsCommand from "../../run/uitest";
+import { Questions } from "inquirer";
+import { directoryExistsSync } from "../../../../util/misc/fs-helper";
+
+interface BuildFolder {
+  name: string;
+  path: string;
+}
+
+export default class RunUitestWizardTestCommand extends AppCommand {
+  private _args: CommandArgs;
+
+  constructor(args: CommandArgs, interactiveArgs: string[]) {
+    super(args);
+    this._args = args;
+    this._args.args.push(...interactiveArgs);
+  }
+
+  public async run(client: AppCenterClient, portalBaseUrl: string): Promise<CommandResult> {
+    const searchFolder: Promise<BuildFolder[]> = this.scanFolder();
+
+    if (this._args.args.indexOf("--async") < 0) {
+      const mergeXml: boolean = await this.promptMergeXml();
+      if (mergeXml) {
+        this._args.args.push("--merge-nunit-xml");
+      }
+    }
+
+    const foundFolders: BuildFolder[] = await searchFolder;
+    const folder: string = await this.promptFolder(foundFolders);
+    this._args.args.push("--build-dir", folder);
+    out.text("\nRunning command: appcenter test run uitest " + this._args.args.join(" ") + "\n");
+    return new RunUITestsCommand(this._args).run(client, portalBaseUrl);
+  }
+
+  private async promptMergeXml(): Promise<boolean> {
+    const questions: Questions = [
+      {
+        type: "list",
+        name: "merge",
+        message: "Should the xml files be merged in to the <output.xml> file?",
+        choices: [{
+          name: "Yes",
+          value: "true"
+        }, {
+          name: "No",
+          value: "false"
+        }]
+      }
+    ];
+    const answers: any = await prompt.question(questions);
+    return answers.merge === "true" ? true : false;
+  }
+
+  private async scanFolder(): Promise<BuildFolder[]> {
+    const foundFolders: BuildFolder[] = [];
+    await this.scanRecurse(process.cwd(), foundFolders);
+    return foundFolders;
+  }
+
+  private async scanRecurse(dirname: string, folders: BuildFolder[]) {
+    const dirContent = fs.readdirSync(dirname);
+    for (const dir of dirContent) {
+      const fullDir = path.join(dirname, dir);
+
+      if (!fs.lstatSync(fullDir).isDirectory()) {
+        continue;
+      }
+      if (dir === "node_modules") {
+        continue;
+      }
+      const dirContents: string[] = fs.readdirSync(fullDir);
+
+      if (dirContents.length === 0) {
+        continue;
+      } else {
```

Comment:
I'd be tempted to leave this `else` out - the `continue` in the if statement will ensure that execution only continues if there are items in `dirContents`. Very minor - just mentioning for completeness! 🙂

Reply:
Yep, indeed :)